#
# cons.saver: open the console in non-blocking mode
#
# Source:	Gentoo Portage
# Reference:	4.6.1
# Submit-By:	Enrico Weigelt, metux IT service <weigelt metux de>
# Submit-Date:	2008-12-26
# State:	new
#
diff -ruN mc-4.6.1.orig/src/cons.saver.c mc-4.6.1/src/cons.saver.c
--- mc-4.6.1.orig/src/cons.saver.c	2008-12-26 02:19:54.000000000 +0100
+++ mc-4.6.1/src/cons.saver.c	2008-12-26 03:15:34.000000000 +0100
@@ -134,7 +134,7 @@
 
   if (seteuid (uid) < 0)
     die ();
-  console_fd = open (tty_name, O_RDONLY);
+  console_fd = open (tty_name, O_RDONLY | O_NONBLOCK);
   if (console_fd < 0)
     die ();
   if (fstat (console_fd, &st) < 0 || ! S_ISCHR (st.st_mode))

